### PR TITLE
Reorganize intro and section 4

### LINKED ISSUE #238: Clarifying Binding Mechanism
From TD and Architecture, we should bring (copyover) the parts that are about the binding mechanism

### LINKED ISSUE #265: Moving types of bindings to section 4
From the call of 15.03:

Would it make more sense to move introduction about different types of bindings to section 4?

So moving [this](https://w3c.github.io/wot-binding-templates/#introduction) to [here](https://w3c.github.io/wot-binding-templates/#binding-overview)

### PR DESCRIPTION
**Pull request by egekorkan**
fixes #265 , fixes #238

Additionally, it does the following:
- Explains when a platform binding makes sense and what it should not do (introduce a new protocol for example)
- Carries over the same structure of sections to platforms
- Copies over content from WoT Architecture on Binding Templates (https://www.w3.org/TR/wot-architecture11/#sec-binding-templates)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-binding-templates/pull/266.html" title="Last updated on Mar 29, 2023, 2:56 PM UTC (33dee71)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-binding-templates/266/c314cbd...33dee71.html" title="Last updated on Mar 29, 2023, 2:56 PM UTC (33dee71)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -157,43 +157,12 @@ <h1>Introduction</h1>
...
-                Description[[WOT-THING-DESCRIPTION]] forms via reusable vocabulary and extensions.</li>
-            <li> <a href="#payload-bindings-intro"></a>: Different payload formats and media types [[IANA-MEDIA-TYPES]] which
-                can be represented in a TD via Data Schemas or forms.</li>
-            <li> <a href="#platforms-intro"></a>: Platforms and frameworks who combine the use of protocols and payloads
-                in a certain way, which can be represented via entire Thing Models described by the [[WOT-THING-DESCRIPTION]]
-                or examples of TDs.</li>
-        </ul>
+             Description[[WOT-THING-DESCRIPTION]] (TD), with Things that exhibit diverse protocols, payload formats and a
+             combination of these in platforms and frameworks.
+             The mechanism that allows this is called the Binding Mechanism, without which TDs could not build Hypermedia Controls
```
Comment (relu91) [inline index.html:162]: *allow this* what?

**Review by relu91 — CHANGES_REQUESTED** [review]
Just two minor points below.

File: index.html (hunk 1 above)
Comment (egekorkan) [reply index.html:162]: I have added some text but it is a bit difficult to express this in a short/elegant way

**Review by relu91 — APPROVED** [review]
I'd say this is a great improvement in the understandability of our binding mechanism. Well done!

Just one minor point, using the verb to `form` in the following sentence feels weird to me:
> This results in different protocols and media types and forms the central challenge for the Web of Things

It might be good to get a native speaker's opinion on this.

**Comment by egekorkan** [comment]
Call of 29.03: No objections, @egekorkan will fix the definition problem and merge

**Pull request merged**